src/main/java/org/stellar/anchor/paymentservice/circle/CirclePaymentsService.java

```diff
@@ -257,6 +260,73 @@ private HttpClient getWebClient(boolean authenticated) {
         return null;
     }
 
+    /**
+     * Validates if the fields needed to send a payment are valid.
+     *
+     * @param sourceAccount      the account where the payment will be sent from.
+     * @param destinationAccount the account that will receive the payment.
+     * @param currencyName       the name of the currency used in the payment. It should obey the {scheme}:{identifier}
+     *                           format described in <a href="https://stellar.org/protocol/sep-38#asset-identification-format">SEP-38</a>.
+     *
+     * @throws HttpException if the source account network is not CIRCLE.
+     * @throws HttpException if the destination account network is not supported.
+     * @throws HttpException if the currencyName prefix does not reflect the destination account network.
+     */
+    private void validateSendPaymentInput(@NonNull Account sourceAccount, @NonNull Account destinationAccount, @NonNull String currencyName) throws HttpException {
+        if (sourceAccount.network != Network.CIRCLE) {
+            throw new HttpException(400, "the only supported network for the source account is circle");
+        }
+        if (!List.of(Network.CIRCLE, Network.STELLAR).contains(destinationAccount.network)) {
+            throw new HttpException(400, "the only supported networks for the destination account are circle and stellar");
+        }
+        if (!currencyName.startsWith(destinationAccount.network.getCurrencyPrefix())) {
+            throw new HttpException(400, "the currency to be sent must contain the destination network schema");
+        }
+    }
```

Comment:
Do we want to validate the sending and receiving account addresses here? 

For example, what if we try to send a payment from Circle to a Stellar account that doesn't exist or doesn't have a trustline? In the opposite direction, what if we try to send to a Circle wallet but the we use a memo that wasn't generated using Circle's `POST /addresses` endpoint?

Reply:
🤔  These are great points! My rationale was to avoid overloading the SDK with unnecessary complexity and simply handle an error when it occurs due to a misconfiguration.

> For example, what if we try to send a payment from Circle to a Stellar account that doesn't exist or doesn't have a trustline? 

That's a great point, depending on how much we want to validate the accounts before sending a payment, this could scale further:
* For a Circle sender, should we validate it exists in the Circle network? Should we verify if the balance is enough?
* For a Circle receiver, should we check if it exists?
* For a Stellar receiver, should we check if it exists and if it has the trustline needed?
* As we expand the support for other networks, we will need to add to these verifications.

I'd address this in another PR if we decide to go through this route.

> In the opposite direction, what if we try to send to a Circle wallet but the we use a memo that wasn't generated using Circle's POST /addresses endpoint?

I don't think this is a use case we need to handle because these memos are used to receive payments from outside Circle. if we need to send a payment from a Circle wallet to another circle wallet, we would use an [internal transfer](https://developers.circle.com/reference#accounts-transfers-create).

Reply:
> I'd address this in another PR if we agree to go through this route.

I think we should eventually, but we can plan to do this later in the package's development.

> I don't think this is a use case we need to handle because these memos are used to receive payments from outside Circle.

I see, so we're not supporting `Stellar->Circle` payments? Why is that?

Reply:
Alright, I created this issue to follow-up on the validation: https://github.com/stellar/java-stellar-anchor-sdk/issues/16

> I see, so we're not supporting Stellar->Circle payments? Why is that?

Let's sync offline about this one, it should be faster.

Reply:
Just to register what we discussed offilne: this PR is restricted to implementing the Circle API integration capable of sending `Circle->Circle` and `Circle->Stellar` transfers.

In the future, we will add a Stellar driver (likely) at `paymentservice.stellar` that's capable of doing `Stellar->Circle` payments.